framework/src/interface/packet.rs

```diff
@@ -400,12 +400,12 @@ impl<T: EndOffset, M: Sized + Send> Packet<T, M> {
        header.size()---NAMES ARE HORRIBLE. In this example, self.offset() +
        self.payload_offset() give you where the tcp payload would start.
 
-       Use *insert_header* on a mutable reference to insert a header into an
+       Use *insert_a_header* on a mutable reference to insert a header into an
        already existing packet, after another header's offset basically. It
        returns a void Result type, i.e. Err or Ok upon inserting a header int
      */
     #[inline]
-    pub fn insert_header<T2: EndOffset<PreviousHeader = T>>(
+    pub fn insert_a_header<T2: EndOffset<PreviousHeader = T>>(
```

Comment:
should this one be non-pub?

Reply:
i still wanted it available for crazy people like us at points :)

Reply:
I'd suggest that until we need it, let's make it private. If we do that, all the fns here (that we wrote at least) will take a good packet and return a good packet, assuming you pass in good values. This thing still only does part of the job. We can always pub it up later if we find a good reason -- though tbh if we find a reason to use this directly, it probably means we need to reconsider the abstractions we've built already, or make a new one :)

Reply:
yeah, ok; can do. i went back and forth on it for a bit, but sure.